torch/utils/data/dataloader.py

```diff
@@ -48,9 +57,46 @@ def _worker_loop(dataset, index_queue, data_queue, collate_fn, seed, init_fn, wo
     if init_fn is not None:
         init_fn(worker_id)
 
+    worker_done_event = threading.Event()
+
+    manager_handle = None
+    if IS_WINDOWS:
+        kernel32 = ctypes.WinDLL('kernel32', use_last_error=True)
+        kernel32.OpenProcess.argtypes = (DWORD, BOOL, DWORD)
+        kernel32.OpenProcess.restype = HANDLE
+        kernel32.WaitForSingleObject.argtypes = (HANDLE, DWORD)
+        kernel32.WaitForSingleObject.restype = DWORD
+
+        # Value obtained from https://msdn.microsoft.com/en-us/library/ms684880.aspx
+        SYNCHRONIZE = 0x00100000
+        manager_handle = kernel32.OpenProcess(SYNCHRONIZE, 0, manager_pid)
+
+        if not manager_handle:
+            raise ctypes.WinError(ctypes.get_last_error())
+
+    def check_manager_process_status(worker_done_event, manager_pid, manager_handle):
```

Comment:
There's no parallelism in checking the status of the main process anyway (because of the GIL), so couldn't we use `index_queue.get()` with a timeout, and just call this function if it times out? There's no need for threads, `worker_done_event`, and the worker can just exit cleanly.

Reply:
It seems that there will be cases where the worker is waiting for the main process to put indices into the `index_queue` (as `_put_indices` is called for every batch), and the wait time is dependent on when `__next__` will be called, so I am not sure what timeout value we should set to differentiate this vs main process crash

Reply:
That's the whole point of checking right? We set a `5s` timeout, if it passes then we check if the main process is alive. If it is, we go back to waiting again, otherwise we exit ourselves.